source/app-development/tutorials-interactive-apps/add-custom-queue/global-static-list.rst

```diff
@@ -0,0 +1,251 @@
+
+#. First we rename the ``form.yml`` for the Jupyter Interactive App to
+   ``form.yml.erb``:
+
+   .. code-block:: shell
+
+      mv ~/ondemand/dev/jupyter/form.yml ~/ondemand/dev/jupyter/form.yml.erb
+
+   This will cause the YAML file to be processed using the `eRuby (embedded
+   Ruby)`_ templating system, which allows us to embed Ruby_ code into the YAML
+   configuration file for flow control, variable substitution, and more.
+
+#. Next we start with the default ``form.yml`` for the Jupyter Interactive App:
+
+   .. code-block:: yaml
+
+      # ~/ondemand/dev/jupyter/form.yml.erb
+      ---
+      cluster: "my_cluster"
+      attributes:
+        modules: "python"
+        conda_extensions: "1"
+        extra_jupyter_args: ""
+      form:
+        - modules
+        - conda_extensions
+        - extra_jupyter_args
+        - bc_account
+        - bc_queue
+        - bc_num_hours
+        - bc_num_slots
+        - bc_email_on_started
+
+#. We remove the following line from this file:
+
+   .. code-block:: yaml
+      :emphasize-lines: 2
+
+      - bc_account
+      - bc_queue
+      - bc_num_hours
+
+   Now when we refresh the web page for our sandbox Jupyter App we won't see
+   the "Queue" form element anymore.
+
+#. We now add in code that reads in the list of available queues/partitions
+   from the cluster configuration file and generates a custom drop-down
+   attribute with this list of queues/partitions:
+
+   .. code-block:: yaml
+      :emphasize-lines: 2-4,11-18,24
+
+      # ~/ondemand/dev/jupyter/form.yml.erb
+      <%-
+        queues = OodAppkit.clusters[:my_cluster].custom_config[:queues]
+      -%>
+      ---
+      cluster: "my_cluster"
+      attributes:
+        modules: "python"
+        conda_extensions: "1"
+        extra_jupyter_args: ""
+        custom_queue:
+          label: Queue
+          help: Please select a queue from the drop-down.
+          widget: select
+          options:
+          <%- queues.each do |q| -%>
+            - [ "<%= q %>", "<%= q %>" ]
+          <%- end -%>
+      form:
+        - modules
+        - conda_extensions
+        - extra_jupyter_args
+        - bc_account
+        - custom_queue
+        - bc_num_hours
+        - bc_num_slots
+        - bc_email_on_started
+
+   At the top we have an *execution tag* that sets a local variable ``queues``
+   from a line of Ruby code that should read in the list of queues you defined
+   under ``custom:`` and ``queues:`` for the corresponding cluster
+   configuration file.
+
+   .. warning::
+
+      The cluster defined in ``OodAppkit.clusters[:my_cluster]`` **must**
+      correspond to a cluster with a cluster configuration file. Also it should
+      match the cluster defined on the line:
+
+      .. code-block:: yaml
+
+         cluster: "my_cluster"
+
+      in the ``form.yml.erb`` file above.
+
+   Later in the YAML file we have another *expression tag* that loops through
```

Comment:
When you say "execution tag" in parens probably should add (i.e. <%- and -%>). Also, I'm confused, does "execution tag" mean `<%-` and "expression tag" mean `<%=`.? In the examples you only use `<%-`.

Also, in documentation I see `<%` but do not see  `<%-`. Why do you use  `<%-`? I know that `%>` and `-%>` differs in that  `-%>` drops trailing whitespace. For example, neither https://ruby-doc.org/stdlib-2.5.0/libdoc/erb/rdoc/ERB.html#method-c-new nor http://api.rubyonrails.org/classes/ActionView/Base.html mention `<%-` but they do mention `-%>` i.e.

> You trigger ERB by using embeddings such as `<% %>`, `<% -%>`, and `<%= %>`.

Reply:
Good points, I should fix those.

> Why do you use `<%-`

This strips leading whitespace:

```ruby
[1] pry(main)> require 'erb'
=> true
[2] pry(main)> ERB.new(" <% true -%>\nword", nil, "-").result
=> " word"
```